client/state/sharing/keyring/actions.js

```diff
@@ -12,9 +17,11 @@ import {
 /**
  * Triggers a network request for a user's connected services.
  *
+ * @param {Function} successCallback Function that is called after a successful
+ *                                   request. Default: _.identity.
  * @return {Function} Action thunk
  */
-export function requestKeyringConnections() {
+export function requestKeyringConnections( successCallback = identity ) {
```

Comment:
For simplicity sake, the equivalent would be a handler firing on each `KEYRING_CONNECTIONS_REQUEST_SUCCESS` dispatch.

However the callback is updating component state rather than triggering another redux action. Can you think of what a similar example might look like with middleware handlers?